apps/desktop-wallet/src/features/historicChart/useHistoricData.tsx

```diff
@@ -0,0 +1,127 @@
+  const allAddressHashes = useAppSelector((s) => s.addresses.ids as AddressHash[])
+  const currency = useAppSelector((s) => s.settings.fiatCurrency).toLowerCase()
+
+  const { data: alphPriceHistory, isPending: isPendingAlphPriceHistory } = useQuery({
+    queryKey: [HISTORY_QUERY_KEY, 'price', ALPH.symbol, { currency }],
+    queryFn: () =>
+      client.explorer.market.getMarketPricesSymbolCharts(ALPH.symbol, { currency }).then((rawHistory) => {
+        const today = dayjs().format(CHART_DATE_FORMAT)
+        const history = [] as TokenHistoricalPrice[]
+
+        if (rawHistory.timestamps && rawHistory.prices) {
+          for (let index = 0; index < rawHistory.timestamps.length; index++) {
+            const timestamp = rawHistory.timestamps[index]
+            const price = rawHistory.prices[index]
+
+            const itemDate = dayjs(timestamp).format(CHART_DATE_FORMAT)
+            const prevItemDate =
+              index > 1 ? dayjs(rawHistory.timestamps[index - 1]).format(CHART_DATE_FORMAT) : undefined
+
+            if (itemDate !== prevItemDate && itemDate !== today) {
+              history.push({
+                date: itemDate,
+                value: price
+              })
+            }
+          }
+        }
+
+        return history
+      })
+  })
+
+  const {
+    data: alphBalanceHistoryPerAddress,
+    isPending: isPendingHistoricalAlphBalances,
+    hasHistoricBalances
+  } = useQueries({
+    queries: allAddressHashes.map((hash) => ({
+      queryKey: [HISTORY_QUERY_KEY, 'addressBalance', DAILY, ALPH.symbol, { hash }],
+      queryFn: async () => {
+        const now = dayjs()
+        const thisMoment = now.valueOf()
+        const oneYearAgo = now.subtract(365, 'days').valueOf()
+
+        const { amountHistory } = await client.explorer.addresses.getAddressesAddressAmountHistory(hash, {
+          fromTs: oneYearAgo,
+          toTs: thisMoment,
+          'interval-type': DAILY
+        })
+
+        return {
+          address: hash,
+          amountHistory
+        }
+      },
+      staleTime: ONE_DAY_MS
+    })),
+    combine: (results) => ({
+      data: results.reduce(
+        (historyPerAddress, { data }) => {
+          if (data) {
+            historyPerAddress[data.address] = !data.amountHistory
+              ? undefined
+              : data.amountHistory.reduce(
+                  (amountPerDate, [timestamp, amount]) => {
+                    const date = dayjs(timestamp).format(CHART_DATE_FORMAT)
+                    amountPerDate[date] = amount
+
+                    return amountPerDate
+                  },
+                  {} as Record<Timestamp, Amount>
+                )
+          }
+
+          return historyPerAddress
+        },
+        {} as Record<AddressHash, Record<Timestamp, Amount> | undefined>
+      ),
+      isPending: results.some(({ isPending }) => isPending),
+      hasHistoricBalances: results.some(({ data }) => data?.amountHistory?.length)
+    })
```

Comment:
Instead of combining everything to arrays, I am combining them to objects. This way, we can access the data faster when we need them in the chart component (see later comment) instead of doing `.find`, which is more expensive. Reduce and map have the same complexity, so here it doesn't cost us anything to return objects instead of arrays.